File: deploy.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const workDir = '/workspaces/PURPOSE-DRIVEN';
const logFile = path.join(workDir, 'deploy-result.log');

function log(message) {
  const line = `${new Date().toISOString()} - ${message}\n`;
  console.log(message);
  fs.appendFileSync(logFile, line);
}

function exec(command) {
  log(`Running: ${command}`);
  try {
    const output = execSync(command, { 
      cwd: workDir, 
      encoding: 'utf8',
      stdio: 'pipe'
    });
    log(`✅ Success: ${output.substring(0, 200)}`);
    return output;
  } catch (error) {
    log(`❌ Error: ${error.message}`);
    if (error.stdout) log(`stdout: ${error.stdout}`);
    if (error.stderr) log(`stderr: ${error.stderr}`);
    throw error;
  }
}

try {
  log('🚀 STARTING DEPLOYMENT');
  
  // Configure git if needed
  try {
    exec('git config user.name');
  } catch (e) {
    exec('git config --global user.name "Copilot Agent"');
    exec('git config --global user.email "copilot@khesed-tek.com"');
  }
  
  // Check status
  log('\n📋 Git Status:');
  const status = exec('git status --short');
  log(status || 'No changes');
  
  // Add files
  log('\n📝 Staging files...');
  exec('git add lib/db.ts lib/auth.ts DEPLOYMENT_AUDIT.md');
  
  // Commit
  log('\n💾 Committing...');
  try {
    exec('git commit -m "CRITICAL: Enhanced database logging & connection testing"');
  } catch (e) {
    if (e.message.includes('nothing to commit')) {
      log('⚠️  Nothing to commit (already committed)');
    } else {
      throw e;
    }
  }
  
  // Push
  log('\n🌐 Pushing to GitHub...');
  const pushOutput = exec('git push origin main');
  log(pushOutput);
  
  log('\n✅ DEPLOYMENT COMPLETE!');
  log('⏳ Vercel will rebuild in 2-3 minutes');
  log('📋 Test at: https://khesed-tek-cms-org.vercel.app/auth/signin');
  
} catch (error) {
  log(`\n❌ DEPLOYMENT FAILED: ${error.message}`);
  process.exit(1);
}

File: scripts/deploy/deploy.py
#!/usr/bin/env python3
import subprocess
import sys
import os

os.chdir('/workspaces/PURPOSE-DRIVEN')

print("🚀 ENTERPRISE DEPLOYMENT PROTOCOL")
print("="*60)

try:
    # Git add
    print("\n📝 Staging changes...")
    result = subprocess.run(['git', 'add', '-A'], capture_output=True, text=True, check=True)
    print("✅ Files staged")
    
    # Git commit
    print("\n💾 Committing...")
    commit_msg = """CRITICAL: Enhanced database logging & connection testing

- Added detailed error logging in lib/auth.ts
- Added pgbouncer validation in lib/db.ts
- Added startup connection test
- Enabled production warning logs
- Removed fallback auth references

AUDIT: Database verified, 2 users confirmed"""
    
    result = subprocess.run(['git', 'commit', '-m', commit_msg], capture_output=True, text=True)
    if result.returncode == 0:
        print("✅ Commit created")
        print(f"   {result.stdout.strip()}")
    elif 'nothing to commit' in result.stdout:
        print("⚠️  Nothing to commit (already committed)")
    else:
        print(f"❌ Commit failed: {result.stderr}")
        sys.exit(1)
    
    # Git push
    print("\n🌐 Pushing to GitHub...")
    result = subprocess.run(['git', 'push', 'origin', 'main'], capture_output=True, text=True, check=True)
    print("✅ Pushed successfully")
    print(f"   {result.stderr.strip()}")
    
    print("\n" + "="*60)
    print("✅ DEPLOYMENT COMPLETE!")
    print("\n⏳ Vercel will rebuild (ETA: 2-3 minutes)")
    print("\n📋 NEXT: Test login at https://khesed-tek-cms-org.vercel.app/auth/signin")
    print("   Credentials: admin@iglesiacentral.com / password123")
    
except subprocess.CalledProcessError as e:
    print(f"\n❌ ERROR: {e}")
    print(f"   stdout: {e.stdout}")
    print(f"   stderr: {e.stderr}")
    sys.exit(1)
except Exception as e:
    print(f"\n❌ UNEXPECTED ERROR: {e}")
    sys.exit(1)
